tracer/src/Datadog.Trace/AppSec/Waf/Waf.cs

```diff
@@ -113,7 +113,11 @@ public bool IsKnowAddressesSuported()
 
         public string[] GetKnownAddresses()
         {
-            return _wafLibraryInvoker.GetKnownAddresses(_wafHandle);
+            _wafLocker.EnterWriteLock();
+            var result = _wafLibraryInvoker.GetKnownAddresses(_wafHandle);
+            _wafLocker.ExitWriteLock();
```

Comment:
This could be try/catched. The Exit should be in a finally clause

Reply:
Sure, I can do that. Still, it will not catch the System.AccessViolationException if it happens. It just crashes the process.

Reply:
Seeing a lock release outside a scope guard just rings my alarms. But, up to you.

Reply:
I have moved the try/catch block one level up. I hope this addresses any concerns.

Reply:
I believe the concern is that if this throws, you'll never release the lock. I would expect something like this:

```csharp
try
{
    _wafLocker.EnterWriteLock();
    var result = _wafLibraryInvoker.GetKnownAddresses(_wafHandle);
}
finally
{
    _wafLocker.ExitWriteLock();
}
```

But then, if you're crashing the process _anyway_ 🤷‍♂️